lib/utils/index.ts

```diff
@@ -79,6 +88,16 @@ type RegExpContextBase = {
      */
     getUsageOfPattern: () => UsageOfPattern
 
+    /**
+     * Returns the capturing group references
+     */
+    getCapturingGroupReferences: () => CapturingGroupReference[]
```

Comment:
These references don't include which capturing group they are about, correct? Are they in the same order as the `getAllCapturingGroups` array?

If yes, then it might be nice to return a `ReadonlyMap<CapturingGroup, CapturingGroupReference>` instead.

Reply:
It currently does not contain information about the regex AST capturing group.
`getAllCapturingGroups` is used to get the count. It is necessary to parse the contents indicated by the argument of the replacer function.

```js
str.replace(/(a)(b)/, (match, p1, p2, offset, string, groups) => {});
str.replace(/(a)(b)(c)/, (match, p1, p2, p3, offset, string, groups) => {});
```

Reply:
Ah, my bad. I didn't think of groups having multiple references.